chia/rpc/wallet_rpc_api.py

```diff
@@ -1233,15 +1234,44 @@ async def verify_signature(self, request) -> EndpointResult:
         :param request:
         :return:
         """
+        input_message: str = request["message"]
+        signing_mode_str: Optional[str] = request.get("signing_mode")
+        # Default to BLS_MESSAGE_AUGMENTATION_HEX_INPUT as this RPC was originally designed to verify
+        # signatures made by `chia keys sign`, which uses BLS_MESSAGE_AUGMENTATION_HEX_INPUT
+        if signing_mode_str is None:
+            signing_mode = SigningMode.BLS_MESSAGE_AUGMENTATION_HEX_INPUT
+        else:
+            try:
+                signing_mode = SigningMode(signing_mode_str)
+            except ValueError:
+                raise ValueError(f"Invalid signing mode: {signing_mode_str}")
```

Comment:
This modification will print the `repr()` of the string making it easier to see special cases like trailing spaces or newlines etc.  I guess it could be argued that converting it to a JSON encoded string might be more language agnostic.

```python-console
>>> s = "abc "
>>> print(f"{s}")
abc 
>>> print(f"{s!r}")
'abc '
```

```suggestion
                raise ValueError(f"Invalid signing mode: {signing_mode_str!r}")
```